chips/imxrt10xx/src/ccm.rs

```diff
@@ -1,44 +1,60 @@
-use kernel::common::registers::{register_bitfields, ReadOnly, ReadWrite};
+use kernel::common::registers::{register_bitfields, register_structs, ReadOnly, ReadWrite};
 use kernel::common::StaticRef;
 use kernel::ClockInterface;
 
-// Clock Controller Module
-#[repr(C)]
-struct CcmRegisters {
-    // CCM control register
-    ccr: ReadWrite<u32, CCR::Register>,
-    _reserved1: [u8; 4],
-    // CCM status register
-    csr: ReadOnly<u32, CSR::Register>,
-    // CCM Clock Switcher Register
-    ccsr: ReadWrite<u32, CCSR::Register>,
-    // ARM clock root
-    cacrr: ReadWrite<u32 /* Unimplemented */>,
-    // Bus clock divider
-    cbcdr: ReadWrite<u32 /* Unimplemented */>,
-    // Bus clock multiplexer
-    cbcmr: ReadWrite<u32 /* Unimplemented */>,
-    // Serial clock multiplexer 1
-    cscmr1: ReadWrite<u32, CSCMR1::Register>,
-    // Serial clock multiplexer 2
-    cscmr2: ReadWrite<u32 /* Unimplemented */>,
-    cscdr1: ReadWrite<u32, CSCDR1::Register>,
-    _reserved3: [u8; 44],
-    clpcr: ReadWrite<u32, CLPCR::Register>,
-    _reserved4: [u8; 16],
-    // clock gating register 0
-    ccgr0: ReadWrite<u32, CCGR0::Register>,
-    // clock gating register 1
-    ccgr1: ReadWrite<u32, CCGR1::Register>,
-    // clock gating register 2
-    ccgr2: ReadWrite<u32, CCGR2::Register>,
-    // clock gating register 3
-    ccgr3: ReadWrite<u32, CCGR3::Register>,
-    // clock gating register 4
-    ccgr4: ReadWrite<u32, CCGR4::Register>,
-    // clock gating register 5
-    ccgr5: ReadWrite<u32, CCGR5::Register>,
-    _reserved6: [u8; 8],
+register_structs! {
+    /// Clock Controller Module
+    CcmRegisters {
+        /// CCM Control Register
+        (0x000 => ccr: ReadWrite<u32, CCR::Register>),
+        (0x004 => _reserved0),
```

Comment:
I think the idea behind the `(<address> => <type>)` syntax is that we don't need the reserved fields here.

Reply:
I thought wrong! :smile: